README.md

```diff
@@ -8,14 +8,157 @@ Based on steverhoades/oauth2-openid-connect-client.
 
 ## Usage
 
-Publish config with artisan command `vender:publish` for provider `LaravelOIDCAuth\OIDCAuthServiceProvider`, and fill it.
+Publish config with artisan command `vender:publish` for provider `LaravelOIDCAuth\OIDCAuthServiceProvider`, and fill
+it.
 
-Use `LaravelOIDCAuth\Authenticate` as auth middleware to redirect directly to OIDC login automatically. Alternatively, you can get login url by yourself and store state string into session storage `oidc-auth.state`, see that middleware.
+Use `LaravelOIDCAuth\Middleware\Authenticate` as auth middleware to redirect directly to OIDC login automatically. Alternatively,
+you can get login url by yourself and store state string into session storage `oidc-auth.state`, see that middleware.
 
 ### `config/auth.php`
 
 Set guard driver to `oidc-auth-session` to remove tokens from session storage on logout.
 
-Set provider driver to `oidc-auth-session` to make `Auth::user()` return a `LaravelOIDCAuth\OIDCUser` authenticable from saved token.
+Set provider driver to `none`, user provider is not suitable.
 
-Alternatively, you can implement a user factory (see `LaravelOIDCAuth\UserFactoryInterface`) for things like creating user DB model and use provider driver like `eloquent`.
+Alternatively, you can implement a user factory (see `LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory`) for things
+like creating user DB model and use provider driver like `eloquent`.
+See [User Model & User Factory](#user-model--user-factory)
+
+## Documentation
+
+### Tokens
+
+You can use authenticated user object to get tokens.
+
+```php
+// Get login user, returns \LaravelOIDCAuth\Contracts\OIDCAuthenticatable|null
+Auth::user();
+// Get the saved access token, returns \OpenIDConnectClient\AccessToken|null
+Auth::user()->accessToken();
+// Get the saved id token, returns \Lcobucci\JWT\Token|null
+Auth::user()->accessToken()->getIdToken();
+// Get the saved refresh token, returns string|null
+Auth::user()->accessToken()->getRefreshToken();
+```
+
+### ID Token
+
+You can get id token claims like this:
+
+```php
+Route::get('/user', function () {
+    return Auth::user()->accessToken()->getIdToken()->getClaims();
+});
+```
+
+### Refresh the access token
+
+If you are using `oidc-auth-session` guard with `OIDCUser`, you can easily refresh the access token:
+
+```php
+use LaravelOIDCAuth\Facades\OIDC;
+// ...
+Auth::user()->refreshAccessToken();
+```
+
+Typically, you don't need to refresh the access token manually, the following doc shows an automatic way to do this.
+
+#### Auto refresh
+
+Sometimes, you want the `oidc-auth-session` to refresh the access token automatically every request, please edit
+`config/oidc_auth.php`:
+
+```php
+    // ...
+    'auto_refresh' => true,
+    // ...
+```
+
+Next time the user logged in, the `CallbackController` will set `rememberMe` option from
```

Comment:
Is this rememberMe mechanism related to refreshing token? Why is it tied?

Reply:
Yes, it is.

If the user log in with 'remember me', guard will automatically refresh the access token to keep the user logged in when access token is expired.  Actually, the 'auto_refresh' option is treated as the default value of the rememberMe argument when controller log the user in(See: [loginUser](https://github.com/zhshize/laravel-oidc-auth/blob/53bfe0bfbfc47104e9a7ef6e1c6618b79fb2991f/src/CallbackController.php#L91)).  This mechanism is a simulation of remember me feature when we integrate OIDC to Laravel authentication.

Do you think the mechanism should be described in the doc or this mechanism could be confusing?

Reply:
Well I see you just removed that in newer commit.

If it is implementation detail, it should not be doc'ed at README. We do not want to gurantee this level of compatibility.

IIRC, "remember me" in Laravel document is some sort of long-lived cookie stuff, which is unrelated to this token refreshing behavior. Better use a more straight-forward naming in code instead of mimicking something that we don't actually implement.

Reply:
I renamed the parameter remember to auto_refresh in the newest commit.